src/many/src/protocol.rs

```diff
@@ -91,27 +64,19 @@ impl Ord for Attribute {
 
 impl Debug for Attribute {
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
-        if let Some(ref arguments) = self.arguments() {
-            f.debug_struct("Attribute")
-                .field("id", &self.id)
-                .field("arguments", &arguments)
-                .finish()
-        } else {
-            f.write_fmt(format_args!("Attribute({})", self.id))
-        }
+        f.debug_struct("Attribute")
+            .field("id", &self.id)
+            .field("arguments", &self.arguments)
+            .finish()
     }
 }
 
 impl Encode for Attribute {
     fn encode<W: Write>(&self, e: &mut Encoder<W>) -> Result<(), Error<W::Error>> {
-        if let Some(arguments) = self.arguments() {
-            e.array(1 + arguments.len() as u64)?;
-            e.u32(self.id as u32)?;
-            for a in arguments {
-                e.encode(a)?;
-            }
-        } else {
-            e.u32(self.id as u32)?;
+        e.array(1 + self.arguments.len() as u64)?;
```

Comment:
If there's no argument, the encoding is still just the attribute number.

Reply:
Good catch! I will add a test.